lib/pbench/server/api/resources/upload_api.py

```diff
@@ -51,69 +51,75 @@ def __init__(self, config, logger):
 
     @Auth.token_auth.login_required()
     def put(self, controller):
-
         try:
             username = Auth.token_auth.current_user().username
         except Exception:
             self.logger.exception("Tarfile upload: Exception verifying the username")
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
-        if not request.headers.get("filename"):
-            self.logger.debug(
-                "Tarfile upload: Post operation failed due to missing filename header"
+        filename_hdr = request.headers.get("filename")
+        if not filename_hdr:
+            self.logger.warning(
+                "Tarfile upload: PUT operation failed due to missing" " filename header"
             )
             abort(
                 400,
-                message="Missing filename header, POST operation requires a filename header to name the uploaded file",
+                message="Missing filename header, PUT operation requires a"
+                " filename header to name the uploaded file",
+            )
+        filename = secure_filename(filename_hdr)
+        if not self.allowed_file(filename):
+            self.logger.warning(
+                f"Tarfile upload: Bad file extension received for file {filename}"
             )
-        filename = secure_filename(request.headers.get("filename"))
+            abort(400, message="File extension not supported. Only .xz")
 
-        if not request.headers.get("Content-MD5"):
-            self.logger.debug(
-                f"Tarfile upload: Post operation failed due to missing md5sum header for file {filename}"
+        md5sum = request.headers.get("Content-MD5")
+        if not md5sum:
+            self.logger.warning(
+                "Tarfile upload: PUT operation failed due to missing md5sum"
+                f" header for file {filename}"
             )
             abort(
                 400,
-                message="Missing md5sum header, POST operation requires md5sum of an uploaded file in header",
+                message="Missing md5sum header, PUT operation requires md5sum"
+                " of an uploaded file in header",
             )
-        md5sum = request.headers.get("Content-MD5")
 
-        if not self.allowed_file(filename):
-            self.logger.debug(
-                f"Tarfile upload: Bad file extension received for file {filename}"
+        content_length_hdr = request.headers.get("Content-Length")
+        if not content_length_hdr:
+            self.logger.warning(
+                "Tarfile upload: No Content-Length header value found for"
+                f" file {filename}"
             )
-            abort(400, message="File extension not supported. Only .xz")
-
+            abort(400, message="Missing required content-length header")
```

Comment:
Interestingly, I found by accident yesterday that there's a "411 Length Required" error defined.

There's also a "413 Payload Too Large"...